ui/markdown-editor.js

```diff
@@ -75,6 +76,180 @@ Vue.component('markdown-editor', {
       this.blobUrlCache[blobId] = blobURL
     },
 
+    suggestChannels: function(searchString, cb) {
+      const searchForChannel = searchString.substring(1, searchString.length)
+      SSB.db.query(
+        and(not(channel('')), type('post'), isPublic()),
```

Comment:
this will look through almost the whole database. Needs some kind of limiter. We might think about doing this once and doing a channel cache. We are doing quite a lot of the same thing in /public

Reply:
Come to think of it, I think most of these filter on channel should use the [channel operator](https://github.com/ssb-ngi-pointer/ssb-db2/blob/e3898ff578021fc8efee935252691be8947a6013/operators/index.js#L48) to search instead of streaming messages.

Reply:
That one might need a:

```
 prefix: 32,
 prefixOffset: 1,
 useMap: true,
```

Reply:
Well, that reply didn't go where I wanted.  And I had a typo.

That one does filter using not(channel('')), but I couldn't find a way to filter on only part of a channel. Or does that just mean that the channel must contain something as opposed to having to match the full string?

I will note, too, that even with that in place I still got results which had channel == null. So there may be other issues with channel().

So far, the places I know that we try to pull a channel list are:

1. Public (for the channel selector when you go to post a new message)
2. Channels tab
3. The suggestion box

I could remove the channel suggestion box for now, if you'd like.  It's not like it's a difficult thing to patch back in later.  I tried to leave this extensible enough for profiles and blobs, and then actually realized I could do channels as somewhat of an afterthought.